src/include/buffer.h

```diff
@@ -318,6 +323,15 @@ namespace buffer CEPH_BUFFER_API {
       void copy(unsigned len, list &dest);
       void copy(unsigned len, std::string &dest);
       void copy_all(list &dest);
+
+      friend bool operator==(const iterator_impl& lhs,
+			     const iterator_impl& rhs) {
+	return lhs.bl == rhs.bl && lhs.off == rhs.off;
```

Comment:
should implement all these in buffer.cc (particularly because this is a public api)